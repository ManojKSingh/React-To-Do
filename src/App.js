import React, { useState } from 'react';
import { ThemeProvider } from 'styled-components';
import ToDoContextProvider from './contexts/ToDoContext';
import themes from './themes';
import {ToDoList} from './components/organisms/ToDoList';

function App() {
  const [isDark, toggleDark] = useState(false);
  const themeType = isDark ? "dark" : "light";
  const theme = themes[themeType];

  const toggleTheme = () => toggleDark(!isDark);

  return (
    <ThemeProvider theme={theme}>
      <ToDoContextProvider>
      <div>
        <ToDoList toggleTheme={toggleTheme}/>
      </div>
      </ToDoContextProvider>
    </ThemeProvider >
  );
}
export default App;
