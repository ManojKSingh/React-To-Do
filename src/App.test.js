import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

test('render app', () => {
  const tree = renderer
    .create(<App/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
