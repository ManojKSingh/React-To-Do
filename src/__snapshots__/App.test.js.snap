// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app 1`] = `
<div>
  <div
    className="sc-AxhUy gLiyzU"
  >
    <div
      className="theme-switch"
    >
      <label
        className="sc-AxhCb cPNfnP switch"
      >
        <input
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="slider round"
        />
      </label>
      Dark Mode
    </div>
    <div
      className="to-do-container"
    >
      <h2>
        To Do List
      </h2>
      <form
        onSubmit={[Function]}
      >
        <input
          className="sc-AxjAm cEocgs to-do-input"
          name="toDo"
          onChange={[Function]}
          type="text"
          value=""
        />
        <button
          className="sc-AxirZ lfnxhL"
          type="submit"
        >
          Add
        </button>
      </form>
      <div />
    </div>
  </div>
</div>
`;
